feat: Implement winston logger S2_US_002_[02], health check S2_US_002_[05] and global error handler S2_US_002_[03]

jjtorres-dev opened the pull request:
implement winston logger, health check and global error handler

📋 Descripción
Implementación completa de tres funcionalidades críticas:

✅ Winston Logger System
- Logs detallados y persistentes con Winston
- Archivos separados (combined.log, error.log)
- Integración con Morgan para HTTP logging
- Consola con colores para desarrollo

✅ Health Check Endpoint  
- Ruta: GET /api/v1/health
- Verificación real del estado de la base de datos
- Retorna 200 con 'connected' cuando BD está activa
- Retorna 503 con 'error' cuando BD está caída

✅ Global Error Handler Middleware
- Manejo centralizado de errores
- Respuestas JSON estandarizadas
- Logging automático de errores con Winston

🎯 User Stories Relacionadas
Closes #15, #16, #18

📝 Tipo de Cambio
- [x] Nueva feature

🧪 Tests
- [x] Todos los tests pasan
- [x] Validación manual de funcionalidad

📋 Checklist
- [x] Mi código sigue las convenciones del proyecto
- [x] He realizado una auto-revisión de mi código
- [x] He comentado mi código, particularmente en áreas difíciles de entender
- [ ] He actualizado la documentación correspondiente
- [x] Mis cambios no generan nuevas advertencias
- [ ] He agregado tests que prueban mi fix o feature
- [x] Tests nuevos y existentes pasan localmente

🏗️ Mejoras de Arquitectura
- Separación de concerns (routes, middlewares, config)
- Logger centralizado en toda la aplicación
- Código organizado y mantenible
- Respuestas JSON estandarizadas

🎯 Criterios de Aceptación Cumplidos
Winston Logger (#15)
- [x] Instalar Winston
- [x] Crear carpeta logs/ (agregada a .gitignore)
- [x] Crear archivo src/config/logger.js
- [x] Instalar Morgan para HTTP logging
- [x] Actualizar app.js para usar logger
- [x] Validar que logs se escriben en archivos
- [x] Validar que en consola aparecen logs con colores

Health Check Endpoint (#18)
- [x] Crear endpoint /api/v1/health
- [x] Verificación real de conexión a BD
- [x] Respuestas adecuadas (200/503)
- [x] Logging con Winston centralizado

Error Handler Middleware (#16)
- [x] Middleware global de errores
- [x] Respuestas JSON formateadas
- [x] Manejo de 404 y errores internos
- [x] Logging centralizado con Winston

🔗 Links Relacionados
- Tickets: #15, #16, #18
- Documentación Winston: https://github.com/winstonjs/winston"

jjtorres-dev committed:
feat: configure winston logger system only

- Install and configure winston with custom levels
- Create logs folder and add to .gitignore
- Implement logger.js with ES modules
- Integrate morgan HTTP logging with winston
- Update app.js to use logger across all endpoints
- Add file logging to combined.log and error.log
- Add colored console output for development
- Implement error handling with automatic logging
- REMOVED: Health check endpoint (will be in separate ticket)
- All Winston acceptance criteria validated and working

jjtorres-dev commented:
@muke78 He completado la implementación del Winston Logger. ¿Podrías revisar el PR cuando tengas momento?

muke78 commented on app.js:
```diff
@@ -1,24 +1,29 @@
-app.use(morgan("dev"));
 
-// Ruta raiz
+app.use(
+	morgan("combined", {
+		stream: {
+			write: (message) => logger.http(message.trim()),
+		},
+	}),
+);
+
```
Hola juan jose, este cambio esta super, genial que no solo sea en dev la configuracion de las respuestas que llegan mediante los http si no que agregue mas cosas, super!

muke78 commented on app.js:
```diff
@@ -30,13 +35,47 @@ app.get("/", (_req, res) => {
+		timestamp: new Date().toISOString(),
+		environment: process.env.NODE_ENV,
+	});
+});
```
Esta parte del codigo que agregaste es otro ticket, esta separado de la configuracion de winston, el ticket aca en Github y en Jira como lo tengo es el 
- #18 

Esta mal ubicada y no debe de estar en el archivo principal debe de estar en la carpeta de `src/routes/health.routes.js` con su centralizacion de rutas en `src/routes/index.js`

Ademas tener en cuenta ya que hiciste el logger en winston, el logger debe de ser una funcion centralizada de todo el codigo, a que me refiero, en donde sea que lo ocupemos no se puede estar repitiendo en todo el archivo ya sea en rutas cuando guardas, cuando actualizas

muke78 commented on app.js:
```diff
@@ -30,13 +35,47 @@ app.get("/", (_req, res) => {
+		path: req.originalUrl,
+		timestamp: new Date().toISOString(),
+	});
+});
```
Esto esta genial! se puede optimizar mas en un futuro

muke78 commented on app.js:
```diff
@@ -30,13 +35,47 @@ app.get("/", (_req, res) => {
+			process.env.NODE_ENV === "development" ? error.message : "Algo salió mal",
+		timestamp: new Date().toISOString(),
+	});
+});
```
De igual manera este es otro ticket, este codigo hace un middleware global de errores, para centralizado de errores globales
- #16 

Este codigo igual debe de estar separado correctamente del archivo principal de rutas y debe de servir para toda la apliacion al momento de manejar los errores de forma global

Ademas tener en cuenta ya que hiciste el logger en winston, el logger debe de ser una funcion centralizada de todo el codigo, a que me refiero, en donde sea que lo ocupemos no se puede estar repitiendo en todo el archivo ya sea en rutas cuando guardas, cuando actualizas

muke78 commented on src/config/logger.js:
```diff
@@ -0,0 +1,55 @@
+
+const logger = winston.createLogger({
+	levels: logLevels,
+	level: process.env.LOG_LEVEL || "info",
```
Al declarar una nueva variable de entorno es necesario que en el archivo de .env.example documentes que se debe de poner al momento de ocupar este logger, como recomendacion esta bien, se estructuro de una buena manera dejando en funciones aparte el formateo y los niveles de este

muke78 commented on .env.example:
Este archivo es de suma importancia y no se debe de borrar, puede que haya habido una confusion, el .env.example se puede subir a github ya que no tiene ninguna de las claves del proyecto solo es por documentacion que se debe de tener en cuenta para manejar el flujo de claves de seguridad

muke78 commented:
Hola @jjtorres-dev 👋 , listo ya hice revision de tus cambios que subiste a github, note que hiciste tres tickets que fueron los siguientes
- #15 
- #16 
- #18 

Te asigne los otros dos tickets que hay, la PR o la solicitud de cambios aun no puede pasar o ser subida, se tiene que hacer un fix de estos cambios, como consejo para llevar un mejor control de el flujo de tickets, hacer uno en cada rama independiente, como extra en cuestion de usabilidad, el formato de las ramas es el siguiente `(funcionalidad a hacer)/descripcion_de_la_funcionalidad_que_se_hara`

Quedo al pendiente de cualquier duds que tengas 😉

jjtorres-dev committed:
refactor: centralize winston logger with helper functions

- Add logInfo, logError, logWarning, logDebug, logHttp helper functions
- Update app.js to use centralized logging functions
- Remove repetitive logger calls throughout the code
- Improve code reusability and maintainability
- All logger functionality now accessible via centralized helpers

jjtorres-dev commented:
@muke78 He completado la centralización del logger:

✅ Implementadas funciones helper: logInfo(), logError(), logWarning(), etc.
✅ Eliminada la repetición de código de logging en todo el proyecto
✅ Código más limpio, mantenible y reutilizable
✅ Todo funcionando correctamente con los logs centralizados

El PR está listo para revisión final.

jjtorres-dev committed:
feat: implement health check endpoint and global error handler

- Implement S2_US_002_[05]: Health Check endpoint
  - Create src/routes/health.routes.js with real DB connection check
  - Add centralized routes in src/routes/index.js
  - Health check returns 200 with 'connected' when DB is active
  - Health check returns 503 with 'error' when DB is down
  - Use centralized Winston logger

- Implement S2_US_002_[03]: Global Error Handler Middleware
  - Create src/middlewares/errorHandler.js with standardized error responses
  - Create src/middlewares/index.js for middleware centralization
  - Update app.js to use centralized error handlers
  - Add 404 not found handler with formatted JSON responses
  - Use centralized Winston logger for all error logging

- Database configuration
  - Create src/config/database.js with MySQL connection pool
  - Fix MySQL2 configuration warnings by removing invalid options

- Code organization
  - Separate concerns: routes, middlewares, and configuration
  - Centralize logging using Winston functions
  - Standardize JSON responses across all endpoints

Tests performed:
- Health check returns proper status when MySQL is connected/disconnected
- Error handler returns formatted JSON for 404 routes
- Global error handler captures and formats all errors
